ios/RNSScreenStack.mm

```diff
@@ -434,33 +458,43 @@ - (void)setPushViewControllers:(NSArray<UIViewController *> *)controllers
   }
 
   UIViewController *top = controllers.lastObject;
-  UIViewController *lastTop = _controller.viewControllers.lastObject;
+#ifdef RN_FABRIC_ENABLED
+  UIViewController *previousTop = _controller.topViewController;
+#else
+  UIViewController *previousTop = _controller.viewControllers.lastObject;
+#endif
 
-  // at the start we set viewControllers to contain a single UIVIewController
+  // At the start we set viewControllers to contain a single UIViewController
   // instance. This is a workaround for header height adjustment bug (see comment
   // in the init function). Here, we need to detect if the initial empty
   // controller is still there
-  BOOL firstTimePush = ![lastTop isKindOfClass:[RNSScreen class]];
+  BOOL firstTimePush = ![previousTop isKindOfClass:[RNSScreen class]];
 
   if (firstTimePush) {
     // nothing pushed yet
     [_controller setViewControllers:controllers animated:NO];
-  } else if (top != lastTop) {
-    // we always provide `animated:YES` since, if the user does not want the animation, he will provide
-    // `stackAnimation: 'none'`, which will resolve in no animation anyways.
-    if (![controllers containsObject:lastTop]) {
+  } else if (top != previousTop) {
+    if (![controllers containsObject:previousTop]) {
       // if the previous top screen does not exist anymore and the new top was not on the stack before, probably replace
       // was called, so we check the animation
-      if (![_controller.viewControllers containsObject:top] &&
-          ((RNSScreenView *)top.view).replaceAnimation == RNSScreenReplaceAnimationPush) {
+#ifdef RN_FABRIC_ENABLED
+      if (![_controller.viewControllers containsObject:top])
+#else
+      if (![_controller.viewControllers containsObject:top] && ((RNSScreenView *)top.view).replaceAnimation == RNSScreenReplaceAnimationPush)
+#endif // RN_FABRIC_ENABLED
```

Comment:
There were different conditions on both archs. I believe second part of condition was not added, because there was not `replaceAnimation` prop back then, but I need confirmation.

Reply:
[e739f8f](https://github.com/software-mansion/react-native-screens/pull/1416/commits/e739f8f9f9d1a2ccec4357b49b8d18f510cb3814)